### PR TITLE
Feat/Article

### PR DESCRIPTION
**Pull request by joaooliveiradev**
- Feat: add a new spacing value and add some final touches on the style of article page and add an article mock in the story of the component
- Refactor: just make onButtonClick prop mandatory
- Feat: add an code component, styles: ad break-all property to break lines in markdown
- Feat: add dynamic import to syntax highlighter
- Feat: add syntax highlighter
- Refactor: remove list style to let ul from markdown with styles
- Styles: remove an style of list's because they are buging the list of markdown
- Feat: add article seo component
- Feat: add article body component and documentation of this component
- Feat: add exact in next seo lock
- Refactor: change some previously packages to exact packages in package json
- Feat: add react markdown to parser markdown to html and rehype sanitize to sanitize the html
- Refactor: remove unused export
- Refactor: again move things of libs folder that are used just in one place to theclosest place that they are called
- Refactor: move things of libs folder that are used just in one place to theclosest place that they are called
- Feat: add robots.txt and sitemap page that generate sitemaps of the application
- Refactor: just rename the env names
- Feat: add application name meta tag
- Refactor: remove wrong cannonical link
- Refactor: remove head component in page, we already use default next seo configs
- Feat: add next-seo, articles generated by gssp, default seo in all the pages and seo for the generated articles
- Feat: add cover image to be our open graph image and add the default seo config to all pages
- Remove the delete file from the exports named file from the libs directory
- Refactor: delete useless file and receive the url path declaring them
- Feat: exporting some components in a named way
- Feat: add next seo library to handle with seo of the pages
- Refactor: change function of reading time and get date to article stats, styles: some incorrect styles fix and feat: add correct tag to some components
- Feat: add documentation for toast component
- Feat: add documentation of error message component and substitute the commom error message that exists in input component
- Feat: add article stats component
- Feat: add article header component
- Feat: add article stats on article card component
- Feat: export the type to type the documentation of storybook
- Feat: add articles documentation component
- Feat: add codec type to article by slug
- Feat: add divider component
- Feat: add article stats component
- Feat: add useGetArticle hook to fetch articles by slug
- Refactor: change articles grid name to articles

### CODE REVIEW

File: src/pages/article/[slug].tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                <Title>{dataOption.value.right.article.title}</Title>
+                <Description>
+                  {dataOption.value.right.article.description}
+                </Description>
+              </HeaderSection>
+              <ArticleBody articleText={dataOption.value.right.article.body} />
+            </Wrapper>
+          </>
+        )
+      } else {
```
Comment (vmarcosp) [inline src/pages/article/[slug].tsx:101]: Esse else e o de baixo não fazem a mesma coisa?

File: src/pages/article/[slug].tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            buttonLabel="Try again"
+            onButtonClick={refetch}
+            isButtonLoading={isFetching}
+            disabled={isFetching}
+          />
+        )
+      }
+    } else {
+      return (
+        <ErrorState
```
Comment (vmarcosp) [inline src/pages/article/[slug].tsx:115]: Aqui também.

File: src/pages/article/[slug].tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+interface GetServerSidePropsParams extends ParsedUrlQuery {
+  slug: string
+}
+
+export const getServerSideProps = async ({
+  params,
+}: GetServerSidePropsContext<GetServerSidePropsParams>) => {
+  const queryClient = new QueryClient()
+
+  const paramsOption = fromNullable(params)
```
Comment (vmarcosp) [inline src/pages/article/[slug].tsx:137]: Porque não usar pipe aqui e na função de baixo?

**Review by vmarcosp — APPROVED** [review]

File: src/pages/article/[slug].tsx (hunk 2 above)
Comment (joaooliveiradev) [reply src/pages/article/[slug].tsx:107]: This comment and the above one i fixed in this [commit](https://github.com/joaooliveiradev/conduit-frontend/pull/40/commits/89cbf2c63a6b6a08dbc6c3c580f802a54d74ef8b)

File: src/pages/article/[slug].tsx (hunk 3 above)
Comment (joaooliveiradev) [reply src/pages/article/[slug].tsx]: What do you think about this [way](https://github.com/joaooliveiradev/conduit-frontend/pull/40/commits/efd2f94c4f8d99167e259430eb2bc7a0d58cfaf2)

**Review by vmarcosp — APPROVED** [review]

**Pull request merged**